lib/PoolConnection.js

```diff
@@ -32,6 +32,13 @@ PoolConnection.prototype.release = function release() {
     return undefined;
   }
 
+  if (this._pool.config.idleConnectionTimeout) {
+    this._idleTimeout = setTimeout(
+      this.destroy.bind(this),
```

Comment:
This should probably end the connection gracefully, running the QUIT packet to the server instead of just dropping the TCP connection. Various database setups will block a client that keeps dropping TCP connections without properly closing them.